src/components/molecules/bar-chart.tsx

```diff
@@ -70,13 +70,12 @@ export const TrackerBarChart: FC<TrackerBarChartProps> = ({
   days = 30,
   yMin = 5,
   ySuffix = '',
-  intervalsCount = 6,
   primaryColor = '#CD4000',
   secondaryColor = '#ACAFC4',
   backgroundColor = colors.white
 }) => {
   const {t} = useTranslation();
-
+  const intervalsCount = axisData.length > 30 ? 7 : 6;
```

Comment:
```suggestion
  const intervalsCount = axisData.length < 30 ? 7 : 6;
```
Issue is with 29 and 30, not 30 and 31